packages/libs/react-ui/src/components/Form/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,88 @@
+import { Button } from '@components/Button/NewButton';
+import type { IInputProps, ITextFieldProps } from '@components/Form';
+import { TextField } from '@components/Form';
+import { ListBox } from '@components/ListBox';
+import { Popover } from '@components/Popover';
+import type { CSSProperties } from 'react';
+import React, { useRef } from 'react';
+import type { AriaComboBoxProps } from 'react-aria';
+import { mergeProps, useComboBox, useFilter } from 'react-aria';
+import { useComboBoxState } from 'react-stately';
+
+export interface IComboBoxProps<T>
+  extends Omit<AriaComboBoxProps<T>, 'id' | 'label'> {
+  id: IInputProps['id'];
+  label: ITextFieldProps['label'];
+  width: CSSProperties['width'];
```

Comment:
why do we need width, you can get the size of the popover from the text input ref and make all the combo box just flow with the content for example when it is inside a flex/grid container